pkg/action/lint.go

```diff
@@ -52,28 +50,19 @@ func NewLint() *Lint {
 
 // Run executes 'helm Lint' against the given chart.
 func (l *Lint) Run(paths []string, vals map[string]interface{}) *LintResult {
-	lowestTolerance := support.ErrorSev
-	if l.Strict {
-		lowestTolerance = support.WarningSev
-	}
-
 	result := &LintResult{}
 	for _, path := range paths {
 		linter, err := lintChart(path, vals, l.Namespace, l.Strict)
 		if err != nil {
-			if err == errLintNoChart {
-				result.Errors = append(result.Errors, err)
-			}
-			if linter.HighestSeverity >= lowestTolerance {
-				result.Errors = append(result.Errors, err)
-			}
-		} else {
-			result.Messages = append(result.Messages, linter.Messages...)
-			result.TotalChartsLinted++
-			for _, msg := range linter.Messages {
-				if msg.Severity == support.ErrorSev {
-					result.Errors = append(result.Errors, msg.Err)
-				}
+			result.Errors = append(result.Errors, err)
+			continue
+		}
+
+		result.Messages = append(result.Messages, linter.Messages...)
+		result.TotalChartsLinted++
+		for _, msg := range linter.Messages {
+			if msg.Severity == support.ErrorSev {
+				result.Errors = append(result.Errors, msg.Err)
```

Comment:
I think we might have a bug that was in the old code that was carried over to the new logic. When `l.Strict` is false `support.ErrorSev` is the error level. When `l.Strict` is true `support.WarningSev` should be used for appending to `result.Errors`.

This is important for the `helm lint` command as the different levels tell it when to return a non-zero exit code to the OS.

In the old code `lowestTolerance` was doing this when an err was returned from `lintChart`.  Could you fix this logic?

For example...
```
lowestTolerance := support.ErrorSev
if l.Strict {
	lowestTolerance = support.WarningSev
}
```

and then later...
```
if msg.Severity >= lowestTolerance {
```

Reply:
Ah sure. I'll add some failing tests and do it. And I did doubt this, but I really didn't know how lint was entirely working. I had read some code inside lint.All and sorts, to understand things like highest severity, not much at the higher level. I better read more code 😅 

So this way we will also be getting warnings along with errors in strict case but only errors in a normal (non strict) case